src/main/java/de/tum/cit/aet/artemis/core/service/cleanup/DataCleanupService.java

```diff
@@ -0,0 +1,155 @@
+     * text blocks, feedback items, and participant scores.
+     *
+     * @param deleteFrom The start of the date range for deleting non-rated results.
+     * @param deleteTo   The end of the date range for deleting non-rated results.
+     * @return a {@link CleanupServiceExecutionRecordDTO} representing the execution record of the cleanup job
+     */
+    public CleanupServiceExecutionRecordDTO deleteNonRatedResults(ZonedDateTime deleteFrom, ZonedDateTime deleteTo) {
+        this.longFeedbackTextRepository.deleteLongFeedbackTextForNonRatedResultsWhereCourseDateBetween(deleteFrom, deleteTo);
+        this.textBlockRepository.deleteTextBlockForNonRatedResultsWhereCourseDateBetween(deleteFrom, deleteTo);
+        this.participantScoreRepository.deleteParticipantScoresForLatestNonRatedResultsWhereCourseDateBetween(deleteFrom, deleteTo);
+        this.participantScoreRepository.deleteParticipantScoresForNonRatedResultsWhereCourseDateBetween(deleteFrom, deleteTo);
+        this.feedbackRepository.deleteOldNonRatedFeedbackWhereCourseDateBetween(deleteFrom, deleteTo);
+        this.resultRepository.deleteNonRatedResultsWhereCourseDateBetween(deleteFrom, deleteTo);
+        return CleanupServiceExecutionRecordDTO.of(this.createCleanupJobExecution(CleanupJobType.NON_RATED_RESULTS, deleteFrom, deleteTo));
+    }
+
+    /**
+     * Deletes rated results, excluding the latest rated result for each participation, for courses conducted within the specified date range.
+     * Also deletes associated long feedback texts, text blocks, feedback items, and participant scores.
+     *
+     * @param deleteFrom The start of the date range for deleting rated results.
+     * @param deleteTo   The end of the date range for deleting rated results.
+     * @return a {@link CleanupServiceExecutionRecordDTO} representing the execution record of the cleanup job
+     */
+    public CleanupServiceExecutionRecordDTO deleteRatedResults(ZonedDateTime deleteFrom, ZonedDateTime deleteTo) {
+        this.longFeedbackTextRepository.deleteLongFeedbackTextForRatedResultsWhereCourseDateBetween(deleteFrom, deleteTo);
+        this.textBlockRepository.deleteTextBlockForRatedResultsWhereCourseDateBetween(deleteFrom, deleteTo);
+        this.feedbackRepository.deleteOldFeedbackThatAreNotLatestRatedResultsWhereCourseDateBetween(deleteFrom, deleteTo);
+        this.participantScoreRepository.deleteParticipantScoresForNonLatestLastResultsWhereCourseDateBetween(deleteFrom, deleteTo);
+        this.participantScoreRepository.deleteParticipantScoresForNonLatestLastRatedResultsWhereCourseDateBetween(deleteFrom, deleteTo);
+        this.resultRepository.deleteNonLatestRatedResultsWhereCourseDateBetween(deleteFrom, deleteTo);
+        return CleanupServiceExecutionRecordDTO.of(this.createCleanupJobExecution(CleanupJobType.RATED_RESULTS, deleteFrom, deleteTo));
+    }
+
+    public List<CleanupServiceExecutionRecordDTO> getLastExecutions() {
```

Comment:
Can you add a JavaDoc?
Additionally the usage of a list seams off to me. How about a Map from CleanupJobType to the last job of that type?